modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala

```diff
@@ -12,42 +13,40 @@ import org.tessellation.schema.trust.{InternalTrustUpdateBatch, PublicTrust, Tru
 object TrustStorage {
 
   def make[F[_]: Monad: Ref.Make]: F[TrustStorage[F]] =
-    for {
-      trust <- Ref[F].of[Map[PeerId, TrustInfo]](Map.empty)
-    } yield make(trust)
-
-  def make[F[_]: Monad](trust: Ref[F, Map[PeerId, TrustInfo]]): TrustStorage[F] =
-    new TrustStorage[F] {
-
-      def updateTrust(trustUpdates: InternalTrustUpdateBatch): F[Unit] =
-        trust.update { t =>
-          t ++ trustUpdates.updates.map { trustUpdate =>
-            val capped = Math.max(Math.min(trustUpdate.trust, 1), -1)
-            val updated = t.getOrElse(trustUpdate.id, TrustInfo()).copy(trustLabel = Some(capped))
-            trustUpdate.id -> updated
-          }.toMap
+    Ref[F]
+      .of[Map[PeerId, TrustInfo]](Map.empty)
+      .map(trust =>
+        new TrustStorage[F] {
+
+          def updateTrust(trustUpdates: InternalTrustUpdateBatch): F[Unit] =
+            trust.update { t =>
+              t ++ trustUpdates.updates.map { trustUpdate =>
+                val capped = Math.max(Math.min(trustUpdate.trust.value, 1), -1)
+                val updated = t.getOrElse(trustUpdate.id, TrustInfo()).copy(trustLabel = capped.some)
+                trustUpdate.id -> updated
+              }.toMap
+            }
+
+          def getTrust: F[Map[PeerId, TrustInfo]] = trust.get
+
+          def getPublicTrust: F[PublicTrust] = getTrust.map { trustValue =>
+            PublicTrust(trustValue.mapFilter(_.publicTrust))
+          }
+
+          def updatePeerPublicTrustInfo(peerId: PeerId, publicTrust: PublicTrust): F[Unit] = trust.update { t =>
+            t.updatedWith(peerId) { trustInfo =>
+              trustInfo
+                .orElse(Some(TrustInfo()))
+                .map(_.copy(peerLabels = publicTrust.labels))
+            }
+          }
+
+          override def updatePredictedTrust(trustUpdates: Map[PeerId, Double]): F[Unit] = trust.update { t =>
```

Comment:
what it overrides if the class extends nothing?

Reply:
Removed.